src/main.rs

```diff
@@ -2519,6 +2519,27 @@ fn run_steps(ib: &mut ImageBuilder) -> Result<()> {
                     &a.publisher,
                 ])?;
             }
+            "pkg_set_p5p_publisher" => {
+                #[derive(Deserialize)]
+                struct PkgSetP5PPublisherArgs {
+                    archive: String,
+                    publisher: String,
+                }
+
+                let a: PkgSetP5PPublisherArgs = step.args()?;
+                let mp = ib.root()?;
+
+                pkg(log, &["-R", &mp.to_str().unwrap(),
+                    "set-publisher", "-p", &a.archive, "--search-first",
+                ])?;
+                pkg(log, &["-R", &mp.to_str().unwrap(), "set-publisher",
+                    "--no-refresh",
+                    "--non-sticky",
+                    &a.publisher,
+                ])?;
```

Comment:
I suspect it would be better to adjust the existing `pkg_set_publisher` step to do this.  You could make the `uri` argument to that step optional, and add a new optional `sticky` boolean that if provided would include an appropriate `--sticky` or `--non-sticky` argument.